ocaml/typing/typecore.ml

```diff
@@ -10004,25 +10006,34 @@ let report_error ~loc env = function
       Location.errorf ~loc ~sub
         "@[This application is complete, but surplus arguments were provided afterwards.@ \
          When passing or calling a local value, extra arguments are passed in a separate application.@]"
-  | Param_mode_mismatch (ty, (_, mkind)) ->
-      let mkind =
+  | Param_mode_mismatch (s, mkind) ->
+      let error_of_equate f (step, {Solver.left; Solver.right}) =
+        let left, right =
+          match (step : equate_step) with
+          | Left_le_right -> left, right
+          | Right_le_left -> right, left
+        in
+        (fun ppf -> f ppf left), (fun ppf -> f ppf right)
```

Comment:
It's a bit weird that `left` and `right` can switch names here. Shouldn't they be called `expected` and `actual` after you match on `step`? That's how they're used in the error message.